src/main/java/com/commercetools/sync/services/ProductTypeService.java

```diff
@@ -105,4 +107,16 @@
     @Nonnull
     CompletionStage<ProductType> updateProductType(@Nonnull final ProductType productType,
                                                    @Nonnull final List<UpdateAction<ProductType>> updateActions);
+
+    /**
+     * Given a productType key, this method fetches a productType that matches this given key in the CTP project defined
+     * in a potentially injected {@link SphereClient}. If there is no matching productType an empty {@link Optional}
```

Comment:
Is there a case when `sphereClient` is not injected?

Reply:
```suggestion
     * in a potentially injected {@link SphereClient}. If there is no matching productType, an empty {@link Optional}
```

Reply:
> Is there a case when sphereClient is not injected?

@lojzatran Not but this is an interface, so it can be easily implemented differently. I didn't want to write a Javadoc which contains too much info from the impl itself.

Reply:
I would then removed `potentially` if it can never be true, so people are not confused.

Reply:
645a13e